src/pybind/util/kaldi_io_pybind.cc

```diff
@@ -30,29 +30,40 @@ void pybind_kaldi_io(py::module& m) {
     using PyClass = Input;
     py::class_<PyClass>(m, "Input")
         .def(py::init<>())
-        .def(py::init<const std::string&, bool*>(),
-             "The normal constructor.  Opens the stream in binary mode. "
-             "Equivalent to calling the default constructor followed by "
-             "Open(); then, if binary != NULL, it calls ReadHeader(), putting "
-             "the output in 'binary'; it throws on error.",
-             py::arg("rxfilename"), py::arg("contents_binary") = nullptr)
-        .def("Open", &PyClass::Open,
-             "Open opens the stream for reading (the mode, where relevant, is "
-             "binary; use OpenTextMode for text-mode, we made this a separate "
-             "function rather than a boolean argument, to avoid confusion with "
-             "Kaldi's text/binary distinction, since reading in the file "
-             "system's text mode is unusual.)  If contents_binary != NULL, it "
-             "reads the binary-mode header and puts it in the  'binary' "
-             "variable.  Returns true on success.  If it returns false it will "
-             "not be open.  You may call Open even if it is already open; it "
-             "will close the existing stream and reopen (however if closing "
-             "the old stream failed it will throw).",
-             py::arg("rxfilename"), py::arg("contents_binary") = nullptr)
-        .def("OpenTextMode", &PyClass::OpenTextMode,
-             "As Open but (if the file system has text/binary modes) opens in "
-             "text mode; you shouldn't ever have to use this as in Kaldi we "
-             "read even text files in binary mode (and ignore the \r).",
-             py::arg("rxfilename"))
+        // the constructor and `Open` both require a `bool*` argument
+        // but pybind11 does not support passing a pointer to a primitive
+        // type, only pointer to customized type is allowed.
+        //
+        // For more information, please refer to
+        // https://github.com/pybind/pybind11/pull/1760/commits/1d8caa5fbd0903cece06ae646447fff9b4aa33c0
+        // https://github.com/pybind/pybind11/pull/1760
+        //
+        // Were it be `bool*`, would it always be non-NULL in C++!
```

Comment:
In comments like this, before you explain *why* this code does what it does, explain what it does.  (I.e. that it returns a pair of bools [opened, binary], where `opened` is True if the stream was successfully opened, and `binary` is True if the the stream was opened and in binary format).  Don't assume that the reader (a) is familiar with pybind11, which is extremely difficult, and (b) has time to dig through Kaldi code to see the type and meaning of Input::Open().

Reply:
@danpovey 
thank you for the comment.

I have changed the documentation.

Since I am lazy to create a local branch so I started to add pybind wrapper for
LatticeArc based on this pullrequest.